R/generate_cpp.R

```diff
@@ -0,0 +1,204 @@
+#' Generate C++ for passing output from model fit into R
+#'
+#' This generates using metadata from `src/model_output.yml`
+#'
+#' @param dest The destination to write generated code to.
+#'
+#' @return Nothing, called to generate code in src dir
+#' @keywords internal
+generate_output_interface <- function(dest) {
+
+  template <- readLines(frogger_file("r_interface/model_output.hpp.in"))
+  output_file <- "model_output.csv"
+  outputs <- utils::read.csv(frogger_file("r_interface", output_file),
+                            colClasses = "character")
+
+  validate_dimensions_columns(colnames(outputs), output_file)
+
+  parsed_outputs <- lapply(seq_len(nrow(outputs)), function(row_num) {
+    row <- outputs[row_num, ]
+    ## When reading csv in excel the header column is included in count
+    csv_row_num <- row_num + 1
+    validate_and_parse_output(as.list(row), output_file, csv_row_num)
+  })
+
+  initialise_r_memory <- vcapply(parsed_outputs, generate_initialise_r_memory)
+  initialise_r_memory <- paste(initialise_r_memory, collapse = "\n")
+  set_r_dimensions <- vcapply(parsed_outputs, generate_set_r_dimensions)
+  set_r_dimensions <- paste(set_r_dimensions, collapse = "\n")
+  copy_data <- vcapply(parsed_outputs, generate_copy_data)
+  copy_data <- paste(copy_data, collapse = "\n")
+  return_output <- vcapply(parsed_outputs, generate_return)
+  return_output <- paste0("return Rcpp::List::create(\n",
+                          paste(return_output, collapse = ",\n"),
+                          ");")
+  header <- paste0(
+    "// Generated by frogger: do not edit by hand\n",
+    "// This file is automatically generated. Do not edit this file. If you ",
+    "want to make changes\n// edit `model_output.hpp.in` and run ",
+    "`./scripts/generate` to regenerate.")
+
+  generated_code <- glue::glue(paste(template, collapse = "\n"),
+                               .open = "{{", .close = "}}")
+  writeLines(generated_code, dest)
+  invisible(TRUE)
```

Comment:
why return this?

Reply:
Actually yeah no good reason, returning the destination now